otter/grade/__init__.py

```diff
@@ -103,6 +106,9 @@ def main(
     if ext not in _ALLOWED_EXTENSIONS:
         raise ValueError(f"Invalid submission extension specified: {ext}")
 
+    if result_queue:
+        loggers.add_queue_handler(result_queue)
```

Comment:
The queue logging handler should be removed when this function exits, in case the same python process calls this function multiple times. I think the best way to accomplish this would be to wrap the function body in a `try` block that removes the logging handler in a `finally` block.

Reply:
I should have been clearer @chrispyles -- so you are cool with how I bring the queue handler into the logger? I will remove all result_queue arguments, and add the try ... finally to remove the queue handler. I got everything into the beta branch. It got confusing because I did not realize you had changed some of my earlier implementations with later commits -- which is fine. I ended up cherry-picking my two commits from the feedback branch and putting them on the beta branch -- It was easier for me to see what was happening and integrate your changes to the files I had touched, and not screw up any of the other changes you made to other files!

I think we are close on all this. Thanks for your feedback.

Reply:
> so you are cool with how I bring the queue handler into the logger

Yes, this approach LGTM.

Reply:
> The queue logging handler should be removed when this function exits, in case the same python process calls this function multiple times. I think the best way to accomplish this would be to wrap the function body in a `try` block that removes the logging handler in a `finally` block.

@chrispyles I think you meant to put the try...finally block around the launch_containers function call in grade/__init__.py. if not tell me. This makes sense though in general.

Reply:
The try/finally should encompass everything from line 111 to at least line 145, assuming that nothing after line 145 generates logs you care about. The reason why is because if an error is thrown before launch_containers is called, we still want to ensure that the logging handler is removed.